tests/test_vert_coords.py

```diff
@@ -0,0 +1,125 @@
+import numpy as np
+import pytest
+
+from pyaerocom.griddeddata import GriddedData
+from pyaerocom.vert_coords import *
+from tests.conftest import lustre_unavail
+from tests.fixtures.tm5 import data_tm5
+
+
+@pytest.fixture
+def alt(data_tm5: GriddedData) -> AltitudeAccess:
+    return AltitudeAccess(data_tm5)
+
+
+@pytest.fixture
+def sin_wave() -> np.ndarray:
+    return np.sin(np.arange(1000))
+
+
+@pytest.mark.parametrize(
+    "sigma,ps,ptop,exception",
+    (
+        pytest.param(1.0, 0, "", ValueError, id="value-error"),
+        pytest.param(1.0, 0, 0.0, None, id="ptop-is-float"),
+        pytest.param(1.0, 0, np.ones(10), None, id="ptop-is-ndarray"),
+    ),
+)
+def test_atmosphere_sigma_coordinate_to_pressure(sigma, ps, ptop, exception):
+    if exception is not None:
+        with pytest.raises(exception):
+            atmosphere_sigma_coordinate_to_pressure(sigma, ps, ptop)
+    else:
+        result = atmosphere_sigma_coordinate_to_pressure(sigma, ps, ptop)
+
+        assert type(result) == type(ptop)
+        if isinstance(result, np.ndarray):
+            assert len(result) == len(ptop)
+
+
+@pytest.mark.parametrize(
+    "a,b,ps,p0,exception",
+    (
+        pytest.param(np.ones(10), np.ones(5), 1, 0, ValueError, id="test-ab-length-mismatch"),
+        pytest.param(np.ones(10), np.ones(10), 1, None, None, id="test-p0-is-none"),
+        pytest.param(np.ones(10), np.ones(10), 1, 1, None, id="test-p0-is-not-none"),
+    ),
+)
+def test_atmosphere_hybrid_sigma_pressure_coordinate_to_pressure(
+    a: np.ndarray, b: np.ndarray, ps: float, p0: float | None, exception
+):
+    if exception is not None:
+        with pytest.raises(exception):
+            atmosphere_hybrid_sigma_pressure_coordinate_to_pressure(a, b, ps, p0)
+    else:
+        result = atmosphere_hybrid_sigma_pressure_coordinate_to_pressure(a, b, ps, p0)
+
+        assert len(a) == len(b) == len(result)
```

Comment:
please consider to split into 2 test functions:
- "happy path" test function: no exceptions are raised; and
- "something went wrong" test function: exceptions are expected